engine/src/main/java/io/zeebe/engine/processing/bpmn/container/CallActivityProcessor.java

```diff
@@ -65,63 +65,38 @@ public void onActivating(final ExecutableCallActivity element, final BpmnElement
         .flatMap(this::checkProcessHasNoneStartEvent)
         .ifRightOrLeft(
             process -> {
-              stateTransitionBehavior.transitionToActivated(context);
+              final var activated = stateTransitionBehavior.transitionToActivated(context);
 
               final var childProcessInstanceKey =
                   stateTransitionBehavior.createChildProcessInstance(process, context);
 
-              final var callActivityInstance = stateBehavior.getElementInstance(context);
-              callActivityInstance.setCalledChildInstanceKey(childProcessInstanceKey);
```

Comment:
Do we not need to set the called child instance key anymore? I see we do it in the applier, but wouldn't that only work if the